sdks/python/apache_beam/yaml/yaml_provider.py

```diff
@@ -383,23 +384,41 @@ def __init__(
     self._jar_provider = jar_provider
     self._classpath = classpath
 
+
   def available(self):
     # pylint: disable=subprocess-run-check
-    trial = subprocess.run(['which', subprocess_server.JavaHelper.get_java()],
```

Comment:
We should fix `subprocess_server.JavaHelper.get_java()` to support windows. This approach effectively removed JAVA_HOME environment handling